tests/profile_check.py

```diff
@@ -0,0 +1,465 @@
+
+    profiles = set()
+    for path in pathlib.Path(profile_dir).iterdir():
+        if path.is_file() and not is_skippable_file(path):
+            profiles.add(path.resolve())
+
+    # Not default, dig deeper
+    if not profiles:
+        nestedDirs = (
+            'groups',
+            'profiles-a-f',
+            'profiles-g-l',
+            'profiles-m-r',
+            'profiles-s-z',
+        )
+        for d in nestedDirs:
+            dirpath = pathlib.Path(pathlib.Path(profile_dir).resolve(), pathlib.Path(d))
+            for p in dirpath.rglob("*"):
+                if p.is_file():
+                    profiles.add(p)
+
+    return profiles
+
+def handleArgs():
+    """DEGRADED are purposed for fatal errors - when the profile set will fail to load entirely"""
+
+    allSeverities = ['DEBUG', 'NOTICE', 'WARNING', 'ERROR', 'CRITICAL', 'DEGRADED']
+    aaRoot = '/etc/apparmor.d'
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-d', '--aa-root-dir', action='store',
+                        default=aaRoot,
+                        help='Target different AppArmor root directory rather than default')
+    parser.add_argument('-p', '--profile', action='append',
+                        help='Handle only specified profile')
+#    parser.add_argument('-s', '--severity', action='append',
+#                        choices=allSeverities,
+#                        help='Handle only specified severity event')
+
+    args = parser.parse_args()
+
+#    if not args.severity:
+#        args.severity = allSeverities
+
+    return args
+
+def main(argv):
+
+    args = handleArgs()
+
+    messages = []
+
+    profile_dir = args.aa_root_dir
+    if not args.profile:
+        profiles = findAllProfileFilenames(profile_dir)
+    else:
+        profiles = set()
+        for p in args.profile:
+            absolutePath = pathlib.Path(p).resolve()
+            profiles.add(absolutePath)
+
+    profile_data = {}
+    for path in sorted(profiles):
+        readApparmorFile_Out = readApparmorFile(path)
+        profilesInFile = readApparmorFile_Out[1]
+        messages.extend(readApparmorFile_Out[0])
+        profile_data.update(profilesInFile)
+
+    for m in messages:
+        m['filename'] = str(m.get('filename'))
+        print(json.dumps(m, indent=2))
+
+    if messages:
+        sys.exit(1)
+
+    return None
+
+if __name__ == '__main__':
+
+    try:
+        from apparmor.regex         import *
+        from apparmor.aa            import is_skippable_file
+        from apparmor.rule.file     import FileRule, FileRuleset
+        from apparmor.common        import convert_regexp
+        try:
+            from apparmor.rule.variable import separate_vars
+        except ModuleNotFoundError:
+            from apparmor.aa            import separate_vars
+ 
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError(f"""Can't find 'python3-apparmor' package! Install with:
+$ sudo apt install python3-apparmor""")
```

Comment:
This should be move outside the main with other import.

See, for example: https://github.com/roddhjav/pass-import/blob/19805430712d3b8b82f58373e3771d21bbe1fa64/pass_import/formats/kdbx.py#L10-L16

Then in the main: https://github.com/roddhjav/pass-import/blob/19805430712d3b8b82f58373e3771d21bbe1fa64/pass_import/formats/kdbx.py#L216